File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n = nums.size();
        int sum = 0;
        int maxSum = INT_MIN;
        for(int i=0;i<n;i++){
            sum+=nums[i];
            if(sum > maxSum) {
                maxSum = sum;
            }
            if(sum < 0){
                sum = 0;
            }
            // sum = max(sum,0);
        }
        return maxSum;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.go
func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}

func maxSubArray(nums []int) int {
    sum := 0
    res := nums[0]

    for i := range nums {
        sum += nums[i]
        res = max(res, sum)
        if sum < 0 {
            sum = 0
        }
    }

    return res
}